x-pack/plugins/security_solution/public/common/containers/first_last_seen/index.tsx

```diff
@@ -10,78 +10,81 @@ import { useCallback, useEffect, useRef, useState } from 'react';
 import { Subscription } from 'rxjs';
 
 import { isCompleteResponse, isErrorResponse } from '@kbn/data-plugin/common';
-import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
-import { useKibana } from '../../../../common/lib/kibana';
-import {
-  HostsQueries,
-  HostFirstLastSeenStrategyResponse,
-  HostFirstLastSeenRequestOptions,
-} from '../../../../../common/search_strategy/security_solution';
 
+import {
+  DocValueFields,
+  Direction,
+  FirstLastSeenQuery,
+  FirstLastSeenRequestOptions,
+  FirstLastSeenStrategyResponse,
+} from '../../../../common/search_strategy';
+import { useAppToasts } from '../../hooks/use_app_toasts';
+import { useKibana } from '../../lib/kibana';
 import * as i18n from './translations';
-import { Direction, DocValueFields } from '../../../../../common/search_strategy';
 
-const ID = 'firstLastSeenHostQuery';
+const ID = 'firstLastSeenQuery';
 
-export interface FirstLastSeenHostArgs {
+export interface FirstLastSeenArgs {
   id: string;
   errorMessage: string | null;
   firstSeen?: string | null;
   lastSeen?: string | null;
   order: Direction.asc | Direction.desc | null;
 }
-interface UseHostFirstLastSeen {
-  docValueFields: DocValueFields[];
-  hostName: string;
-  indexNames: string[];
+interface UseFirstLastSeen {
+  field: string;
+  value: string;
   order: Direction.asc | Direction.desc;
+  defaultIndex: string[];
+  docValueFields: DocValueFields[];
 }
 
-export const useFirstLastSeenHost = ({
-  docValueFields,
-  hostName,
-  indexNames,
+export const useFirstLastSeen = ({
+  field,
+  value,
   order,
-}: UseHostFirstLastSeen): [boolean, FirstLastSeenHostArgs] => {
-  const { data } = useKibana().services;
+  defaultIndex,
+  docValueFields,
+}: UseFirstLastSeen): [boolean, FirstLastSeenArgs] => {
+  const { search } = useKibana().services.data;
   const abortCtrl = useRef(new AbortController());
   const searchSubscription$ = useRef(new Subscription());
   const [loading, setLoading] = useState(false);
-  const [firstLastSeenHostRequest, setFirstLastSeenHostRequest] =
-    useState<HostFirstLastSeenRequestOptions>({
-      defaultIndex: indexNames,
-      docValueFields: docValueFields ?? [],
-      factoryQueryType: HostsQueries.firstOrLastSeen,
-      hostName,
-      order,
-    });
 
-  const [firstLastSeenHostResponse, setFirstLastSeenHostResponse] = useState<FirstLastSeenHostArgs>(
-    {
-      order: null,
-      firstSeen: null,
-      lastSeen: null,
-      errorMessage: null,
-      id: ID,
-    }
-  );
+  const [firstLastSeenRequest, setFirstLastSeenRequest] = useState<FirstLastSeenRequestOptions>({
+    defaultIndex,
+    docValueFields: docValueFields ?? [],
```

Comment:
What are the `docValueFields` needed for? just to understand, sorry for my ignorance 😅